File: src/main/java/ru/spbstu/spartamonitor/SpartaMonitorApp.java
package ru.spbstu.spartamonitor;

import javafx.animation.KeyFrame;
import javafx.animation.Timeline;
import javafx.application.Application;
import javafx.application.Platform;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.stage.Stage;
import javafx.util.Duration;
import ru.spbstu.spartamonitor.eventbus.EventBusFactory;
import ru.spbstu.spartamonitor.events.DrawEvent;
import ru.spbstu.spartamonitor.listeners.DrawListener;
import ru.spbstu.spartamonitor.listeners.ParserListener;

import java.io.IOException;

public class SpartaMonitorApp extends Application {

    private SpartaMonitorController controller;
    private static final int drawTimeOut = 200;

    @Override
    public void start(Stage stage) throws IOException {
        FXMLLoader fxmlLoader = new FXMLLoader(SpartaMonitorApp.class.getResource("sparta-monitor-view.fxml"));
        Scene scene = new Scene(fxmlLoader.load());
        stage.setTitle("Sparta monitor");
        stage.setScene(scene);
        stage.show();

        this.controller = fxmlLoader.getController();
        this.controller.setMainStage(stage);

        EventBusFactory.getEventBus().register(new ParserListener(this.controller));
        EventBusFactory.getEventBus().register(new DrawListener(this.controller));

        Timeline timeline = new Timeline(
                new KeyFrame(
                        Duration.seconds(0),
                        event -> EventBusFactory.getEventBus().post(new DrawEvent(null))
                ),
                new KeyFrame(Duration.millis(drawTimeOut))
        );
        timeline.setCycleCount(Timeline.INDEFINITE);
        timeline.play();
    }

    @Override
    public void stop() {
        this.controller.onClose();
        Platform.exit();
    }

    public static void main(String[] args) {
        launch();
    }
}

File: src/main/kotlin/ru/spbstu/spartamonitor/SpartaMonitorApp.kt
package ru.spbstu.spartamonitor

import javafx.animation.KeyFrame
import javafx.animation.Timeline
import javafx.application.Application
import javafx.application.Application.launch
import javafx.application.Platform
import javafx.fxml.FXMLLoader
import javafx.scene.Scene
import javafx.stage.Stage
import javafx.util.Duration
import ru.spbstu.spartamonitor.eventbus.EventBusFactory
import ru.spbstu.spartamonitor.events.DrawEvent
import ru.spbstu.spartamonitor.listeners.DrawDensityListener
import ru.spbstu.spartamonitor.listeners.DrawListener
import ru.spbstu.spartamonitor.listeners.ParserListener

class SpartaMonitorApp : Application() {
    private val drawTimeout = 200
    private lateinit var controller: SpartaMonitorController

    override fun start(stage: Stage) {
        val fxmlLoader = FXMLLoader(SpartaMonitorApp::class.java.getResource("sparta-monitor-view.fxml"))
        val scene = Scene(fxmlLoader.load())
        stage.title = "Sparta monitor"
        stage.setScene(scene)
        stage.show()

        this.controller = fxmlLoader.getController()
        this.controller.setStageMain(stage)

        EventBusFactory.eventBus.register(ParserListener(this.controller))
        EventBusFactory.eventBus.register(DrawListener(this.controller))
        EventBusFactory.eventBus.register(DrawDensityListener(this.controller))

        val timeline = Timeline(
            KeyFrame(
                Duration.seconds(0.0),
                { EventBusFactory.eventBus.post(DrawEvent(null)) }
            ),
            KeyFrame(Duration.millis(drawTimeout.toDouble()))
        )
        timeline.cycleCount = Timeline.INDEFINITE
        timeline.play()
    }

    override fun stop() {
        this.controller.onClose()
        Platform.exit()
    }
}

fun main() {
    launch(SpartaMonitorApp::class.java)
}